pyserini/search/_dsearcher.py

```diff
@@ -0,0 +1,198 @@
+from typing import List, Optional
+import faiss
+from dataclasses import dataclass
+import numpy as np
+from tqdm import tqdm
+import tensorflow.compat.v1 as tf
+
+
+@dataclass
```

Comment:
Do we need both sparse and dense versions of these? both results and document.

Reply:
I am a bit unsure, since sparse depends on anserini's java class, but it's not necessary for dense